File: kolko&krzyzykpython/game.py
from player import Player
from board import Board

class Game:
    def __init__(self):
        self.player_x = Player("Gracz X", "X")
        self.player_o = Player("Gracz O", "O")
        self.current_player = self.player_x
        self.board = Board()

    def switch_turn(self):
        self.current_player = self.player_o if self.current_player == self.player_x else self.player_x

    def reset_game(self):
        self.board.reset()
        self.current_player = self.player_x

    def reset_scores(self):
        self.player_x.score = 0
        self.player_o.score = 0

File: kolko&krzyzykcs/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace kolko_krzyzykcs
{
    public class Game
    {
        public Player PlayerX { get; set; }
        public Player PlayerO { get; set; }
        public Player CurrentPlayer { get; set; }
        public Board GameBoard { get; set; }

        public Game()
        {
            PlayerX = new Player("Gracz X", "X");
            PlayerO = new Player("Gracz O", "O");
            CurrentPlayer = PlayerX;
            GameBoard = new Board();
        }

        public void SwitchTurn()
        {
            CurrentPlayer = (CurrentPlayer == PlayerX) ? PlayerO : PlayerX;
        }

        public void ResetGame()
        {
            GameBoard.Reset();
            CurrentPlayer = PlayerX;
        }
        public void ResetScores()
        {
            PlayerX.Score = 0;
            PlayerO.Score = 0;
        }

    }

}
